teerex/ias-verify/src/lib.rs

```diff
@@ -35,16 +38,98 @@ mod utils;
 
 const SGX_REPORT_DATA_SIZE: usize = 64;
 #[derive(Encode, Decode, Copy, Clone, TypeInfo)]
+#[repr(C)]
 pub struct SgxReportData {
 	d: [u8; SGX_REPORT_DATA_SIZE],
 }
 
 #[derive(Encode, Decode, Copy, Clone, TypeInfo)]
+#[repr(C)]
 pub struct SGXAttributes {
 	flags: u64,
 	xfrm: u64,
 }
 
+#[derive(Decode, Clone, TypeInfo)]
+#[repr(C)]
+pub struct DcapQuote {
+	header: DcapQuoteHeader,
+	body: SgxReportBody,
+	signature_data_len: u32,
+	quote_signature_data: EcdsaQuoteSignature,
+}
+
+#[derive(Encode, Decode, Copy, Clone, TypeInfo)]
+#[repr(C)]
+pub struct DcapQuoteHeader {
```

Comment:
I wow, I did not know about mobile coin. They made some pretty impressive rust stuff around the intel SDK. They did something that I always had in mind; abstracting away make from the build process.

Anyhow, regarding the types, we have a dealbreaker, as they do not derive the scale codec, which we need.

Reply:
The crate certainly looks interesting. I would suggest to open a ticket to look into it a bit more. During implementation I thought it would also be nice if we could separate the whole attestation code more from the substrate, such that it works without dependencies to substrate. Then we could maybe manage without the scale codec. Maybe that could be another ticket?